### ISSUE TITLE
Random Teleport (/cmi rt) is super duper slow....

### ISSUE THREAD

**Issue by digitalcrowd** [Zrips/CMI#2104]
This seemed to happen back when you changed something in relation to how Random Teleport works on 1.14 servers. It went from being nearly instant to taking anywhere from legit, 20-50 seconds to teleport someone even when TPS is at 19. I have had to switch to BetterRTP since the functionality has been broke. 

---
[19:47:53 INFO]: --------------------------------------------------
[19:47:53 INFO]: CMI plugin version: 8.6.7.5-> 8.6.8.0
[19:47:53 INFO]: Server version: Paper(146) 1.14.4-R0.1-SNAPSHOT
[19:47:53 INFO]: Vault version: 1.7.3-b${env.TRAVIS_BUILD_NUMBER}
[19:47:53 INFO]: --------------------------------------------------

**Comment by Zrips** [comment]
Some extra information about this would be nice, like your config section for that particular world. And id your world pre generated? Does your players have safe teleport permission?

**Comment by digitalcrowd** [comment]
I am using WorldBorder... size is 28,000 blocks wide and tall. All chunks have been pre-generated. The server has 32 cores, 128gb of RAM and runs on NVMe drives. So it is not an issue. BetterRTP has instant RTP with no issue.

World5 is the only one with rt enabled.

cmi.safeteleport is not added to permissions for any players.

`  Worlds:
    # World name to use this feature. Add annother one with appropriate name to enable random teleportation
    world3:
      Enabled: false
      # Max coordinate to teleport, setting to 1000, player can be teleported between -1000 and 1000 coordinates
      MaxRange: 7000
      # If maxcord set to 1000 and mincord to 500, then player can be teleported between -1000 to -500 and 1000 to 500 coordinates
      MinRange: 500
      CenterX: 0
      CenterZ: 0
      Circle: false
      IgnoreWater: false
      IgnoreLava: false
      minY: 0
      maxY: 256
    world5:
      Enabled: true
      MaxRange: 12000
      MinRange: 2000
      CenterX: 0
      CenterZ: 0
      Circle: false
      IgnoreWater: true
      IgnoreLava: true
      minY: 0
      maxY: 256
  # How long force player to wait before using command again.
  Cooldown: 60
  # How many times to try find correct location for teleportation.
  # Keep it at low number, as player always can try again after delay
  MaxTries: 20
  # List of biomes to exclude from random teleportation
  ExcludedBiomes:
  - Ocean
  - Deep ocean
`

**Comment by Zrips** [comment]
Ok, with next build there should be some performance boost for this command.

**Comment by digitalcrowd** [comment]
This issue remains...  it seems to take longer now.

First time I did it, it took 70 seconds, the second time I did it, 90 seconds.

The odd thing is, I issued another /rt command about 80 seconds into the second /rt, wondering if perhaps I just didn't type it, but then about 10 seconds later it sent me to a random location, but then just a few seconds later, sent me to a second /rt location.

It seems like the command is getting sent to some queue that isn't processed very often, and it seems like with more players on, it takes longer. 

The above was done with 18 players online, TPS was around 15 at the time (thanks to 1.14)

[11:20:06 INFO]: --------------------------------------------------
[11:20:06 INFO]: CMI plugin version: 8.6.8.4
[11:20:06 INFO]: Server version: Paper(154) 1.14.4-R0.1-SNAPSHOT
[11:20:06 INFO]: Vault version: 1.7.3-b${env.TRAVIS_BUILD_NUMBER}
[11:20:06 INFO]: --------------------------------------------------
